client/src/api/schema/schema.ts

```diff
@@ -5483,6 +5483,15 @@ export interface components {
              */
             url: string;
         };
+        /** GroupUpdatePayload */
+        GroupUpdatePayload: {
+            /** name of the group */
+            name?: string | null;
+            /** role IDs */
+            role_ids?: string[] | null;
```

Comment:
I assume null should mean to modification should be made, that probably needs to be reflected in the code ?

Reply:
So, I assume no provided value in the payload is "leave it untouched" and explicit provided null or None means "remove all associations". Is that so?

I guess we need to change the old API behavior, it seems this `delete_existing_assocs=False` was preventing removing associations, only new ones could be added, right? :thinking: 

https://github.com/galaxyproject/galaxy/blob/d2956695024812b21898869f3f2127e882365f55/lib/galaxy/managers/groups.py#L95

Reply:
>  explicit provided empty list means "remove all associations". Is that so?

That's a good question, I guess removing associations by omission is a bit of a breaking change 🤔. I'm sorry I missed your question yesterday.

> I guess we need to change the old API behavior, it seems this delete_existing_assocs=False was preventing removing associations

It would on the other hand make it so that not providing the role_ids parameter no change is made, which was my concern with https://github.com/galaxyproject/galaxy/pull/18374#discussion_r1636318636

Reply:
No worries, I reverted the behavior back to keep all existing associations. The fact that once you add a user or role you cannot remove or replace it anymore still feels like a bug in the API and is not consistent with what the `manage_users_and_roles_for_group` admin controller endpoint does. But I understand this will be a breaking change.

I added more detailed checks to the tests to ensure nothing gets changed when it is not explicitly provided in the request.